programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -658,21 +668,43 @@ static void on_event(const base_event *ev) {
             identity_event id_event = {0};
             id_event.Op = strdup("identity");
             id_event.Action = strdup(event_name(event_added));
-            id_event.Id = get_tunnel_identity(ev->identifier);
+            id_event.Id = create_or_get_tunnel_identity(ev->identifier, NULL);
+            id_event.Fingerprint = strdup(id_event.Id->FingerPrint);
             id_event.Id->Loaded = true;
 
             action_event controller_event = {0};
             controller_event.Op = strdup("controller");
             controller_event.Identifier = strdup(ev->identifier);
+            controller_event.Fingerprint = strdup(id_event.Id->FingerPrint);
 
             if (zev->code == ZITI_OK) {
                 id_event.Id->Active = true; // determine it from controller
                 if (zev->name) {
-                    id_event.Id->Name = strdup(zev->name);
-                    if (zev->version != NULL) {
+                    if (id_event.Id->Name != NULL && strcmp(id_event.Id->Name, zev->name) != 0) {
+                        if (id_event.Id->Name) free(id_event.Id->Name);
```

Comment:
there is no need to test `id_event.Id->Name` here, as it was tested on the previous line